packages/alfa-dom/test/node.spec.tsx

```diff
@@ -61,3 +63,75 @@ test(`#tabOrder() correctly handles shadow roots with slotted elements after the
 
   t.deepEqual([...div.tabOrder()], [b, a]);
 });
+
+test(`Node.clone() creates new instance with same value`, (t) => {
+  const device = Device.standard();
+  const doc = h.document(
+    [
+      <p title="foo" box={{ device, x: 1, y: 2, width: 3, height: 4 }}>
+        hello
+      </p>,
+    ],
+    [h.sheet([h.rule.style("p", { background: "green" })])],
+    "bar",
+    { extraStuff: "baz" },
+  );
+
+  const clonedDoc = Node.clone(doc);
+
+  t.notEqual(clonedDoc, doc);
+
+  t.deepEqual(clonedDoc.toJSON(), doc.toJSON());
+});
+
+test(`Node.clone() clones shadow`, (t) => {
+  const div = <div>hello</div>;
+  const shadow = h.shadow([<div>foo</div>]);
```

Comment:
**Note:** Our JSX syntax (actually, `h`) will do shadow magic upon encountering a children that is a shadow tree. So:
```
const div = <div>{shadow}</div>
```
should be enough.
https://github.com/Siteimprove/alfa/blob/main/packages/alfa-dom/src/h.ts#L103-L127
https://github.com/Siteimprove/alfa/blob/main/packages/alfa-dom/test/h.spec.ts#L39

Reply:
Done